Upgrade to Discord.js v13

HoeenCoder opened the pull request:
Upgrades Discord.js to v13, which enables new features and gets us on the latest API version (which also fixes at least 1 bug).

- [ ] Unit tests need to be fixed/redesigned.

HoeenCoder committed:
Start upgrading to Discord.js v13

AnnikaCodes commented on src/client.ts:
```diff
@@ -219,18 +234,18 @@ client.on('message', (m) => void (async msg => {
 		await cmd.execute();
 	} catch (e) {
-		await onError(e, 'A chat command crashed: ');
+		await onError((e as Error), 'A chat command crashed: ');
```
Can you do `catch (e: Error)`, or does TS force you to cast to `unknown` or `any`?

HoeenCoder replied:
Forced type because of TS yeah, its annoying.
![image](https://user-images.githubusercontent.com/11671268/137533366-1a0d7caf-aabe-46b3-bc03-3ee38eee27a9.png)

AnnikaCodes commented on src/command_base.ts:
```diff
@@ -271,6 +271,12 @@ export abstract class BaseCommand {
 		return channel.send('\u274C ' + msg);
 	}
 
+	protected embedReply(embeds: Discord.MessageEmbedOptions[], channel?: DiscordChannel):
+	Promise<Discord.Message> | void {
```
```suggestion
	protected embedReply(
		embeds: Discord.MessageEmbedOptions[],
		channel?: DiscordChannel
	): Promise<Discord.Message> | void {
```
IMO this is cleaner

AnnikaCodes commented on src/events.ts:
```diff
@@ -56,15 +56,14 @@ client.on('messageDelete', (oldMsg: Discord.Message | Discord.PartialMessage) =>
 
 		const log = await fetchAuditLog('MESSAGE_DELETE', oldMessage.guild);
-		if (!log || log.executor.id === oldMessage.author.id) return; // Not a mod delete
+		if (!log || !log.executor || log.executor.id === oldMessage.author.id) return; // Not a mod delete or unable to tell
```
```suggestion
		if (!log?.executor || log.executor.id === oldMessage.author.id) return; // Not a mod delete or unable to tell
```

AnnikaCodes commented on src/monitors/rmt.ts:
```diff
@@ -74,9 +75,9 @@ export class TeamRatingMonitor extends BaseMonitor {
-				if (!user || user.presence.status === 'offline') return false;
+			res = res.filter(async (r) => {
+				const user = await this.guild?.members.fetch(r.userid);
+				if (!user || !user.presence || user.presence.status === 'offline') return false;
```
```suggestion
				if (!user?.presence || user.presence.status === 'offline') return false;
```

AnnikaCodes commented on src/monitors/rmt.ts:
```diff
@@ -74,9 +75,9 @@ export class TeamRatingMonitor extends BaseMonitor {
-			res = res.filter(r => {
-				const user = this.getUser(r.userid);
-				if (!user || user.presence.status === 'offline') return false;
+			res = res.filter(async (r) => {
```
I don't think `Array#filter` will [handle the async callback properly](https://stackoverflow.com/questions/47095019/how-to-use-array-prototype-filter-with-async); it's probably most efficient to just use a `for` loop here and splice out the offline users.

AnnikaCodes reviewed (commented):
When I have some time I'll look into fixing unit tests.

HoeenCoder committed:
Fix build for v13

HoeenCoder committed:
Disable unit tests temporarily

The test suite was broken by the v13 update, and considering v13 is
going to involved a large number of changes to the bot,
its best to just wait to fix unit tests until later.
